c++/src/Timezone.cc

```diff
@@ -656,12 +656,21 @@ namespace orc {
     epoch_ = utcEpoch - getVariant(utcEpoch).gmtOffset;
   }
 
-  const char* getTimezoneDirectory() {
+  std::string getTimezoneDirectory() {
     const char* dir = getenv("TZDIR");
     if (!dir) {
-      dir = DEFAULT_TZDIR;
+      // this is present if we're in an activated conda environment
+      const char* condaPrefix = getenv("CONDA_PREFIX");
+      if (condaPrefix) {
+        std::string condaDir(condaPrefix);
+        condaDir += "/share/zoneinfo";
+        return condaDir;
+      } else {
+        dir = DEFAULT_TZDIR;
+      }
     }
-    return dir;
+    std::string tzDir(dir);
+    return tzDir;
```

Comment:
```suggestion
    return dir;
```
This is enough

Reply:
Yeah, going from `char*` to `std::string` works; I was trying to make the conversion obvious, but happy to do it like this.